File: src/main/java/me/devsnox/playtime/connection/SpigotConnection.java
package me.devsnox.playtime.connection;

import me.devsnox.playtime.configuration.ConnectionConfig;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;

/**
 * Created by Yasin Dalal
 * Copyright (c) 2017-2019 Yasin Dalal
 * GitHub: https://github.com/DevSnox
 * E-Mail: yasin@dalal.ch
 */
@Deprecated
public class SpigotConnection {
    private String host;
    private int port;
    private String database;
    private String username;
    private String password;

    private SyncMySQL syncMySQL;

    private Connection connection;


    public SpigotConnection(String host, int port, String database, String username, String password) {
        this.host = host;
        this.port = port;
        this.database = database;
        this.username = username;
        this.password = password;
        connect();
        this.syncMySQL = new SyncMySQL(this, connection);
    }

    public SpigotConnection(ConnectionConfig config) {
        this.host = config.getHost();
        this.port = config.getPort();
        this.database = config.getDatabase();
        this.username = config.getUsername();
        this.password = config.getPassword();

        connect();

        this.syncMySQL = new SyncMySQL(this, connection);
    }

    public void connect() {
        try {
            this.connection = DriverManager.getConnection("jdbc:mysql://" + this.host + ":" + this.port + "/" + this.database + "?autoReconnect=true", this.username, this.password);
        } catch (SQLException e) {
            e.printStackTrace();
        }
    }

    public void disconnect() {
        try {
            this.connection.close();
        } catch (SQLException e) {
            e.printStackTrace();
        }
    }

    public boolean isConnected() {
        return this.connection != null;
    }

    public SyncMySQL sync() {
        try {
            if (!this.connection.isValid(2)) {
                disconnect();
                connect();
            }
        } catch (SQLException e) {
            disconnect();
            connect();
        }
        return syncMySQL;
    }
}

File: src/main/java/me/devsnox/playtime/connection/SpigotConnection.kt
package me.devsnox.playtime.connection

import me.devsnox.playtime.configuration.ConnectionConfig
import java.sql.Connection
import java.sql.DriverManager
import java.sql.SQLException

/**
 * Created by Yasin Dalal
 * Copyright (c) 2017-2019 Yasin Dalal
 * GitHub: https://github.com/DevSnox
 * E-Mail: yasin@dalal.ch
 */

@Deprecated("Will be replaced by Exposed")
class SpigotConnection {
    private var host: String
    private var port: Int
    private var database: String
    private var username: String
    private var password: String
    private var syncMySQL: SyncMySQL
    private var connection: Connection? = null

    constructor(host: String, port: Int, database: String, username: String, password: String) {
        this.host = host
        this.port = port
        this.database = database
        this.username = username
        this.password = password
        connect()
        syncMySQL = SyncMySQL(this, connection)
    }

    constructor(config: ConnectionConfig) {
        host = config.host
        port = config.port
        database = config.database
        username = config.username
        password = config.password
        connect()
        syncMySQL = SyncMySQL(this, connection)
    }

    fun connect() {
        try {
            connection = DriverManager.getConnection("jdbc:mysql://" + host + ":" + port + "/" + database + "?autoReconnect=true", username, password)
        } catch (e: SQLException) {
            e.printStackTrace()
        }
    }

    fun disconnect() {
        try {
            connection!!.close()
        } catch (e: SQLException) {
            e.printStackTrace()
        }
    }

    val isConnected: Boolean
        get() = connection != null

    fun sync(): SyncMySQL {
        try {
            if (!connection!!.isValid(2)) {
                disconnect()
                connect()
            }
        } catch (e: SQLException) {
            disconnect()
            connect()
        }
        return syncMySQL
    }
}